cfme/tests/v2v/test_v2v_migrations_ui.py

```diff
@@ -241,3 +243,69 @@ def test_v2v_ui_set2(appliance, v2v_providers, form_data_single_datastore, soft_
     soft_assert(set(view.migration_request_details_list.read()) == set(vm_selected))
     view = navigate_to(infrastructure_mapping_collection, 'All')
     view.migration_plans_not_started_list.migrate_plan(plan_name)
+
+    # Test Archive completed migration  plan
+    wait_for(func=view.plan_in_progress, func_args=[plan_name], delay=5, num_sec=300,
+        handle_exception=True)
+    view.migr_dropdown.item_select("Completed Plans")
+    view.wait_displayed()
+    view.migration_plans_completed_list.archive_plan(plan_name)
+    view.migr_dropdown.item_select("Archived Plans")
+    view.wait_displayed()
+    soft_assert(plan_name in view.migration_plans_archived_list.read())
+
+
+@pytest.mark.parametrize('form_data_multiple_vm_obj_single_datastore', [['nfs', 'nfs',
+    [rhel7_minimal, rhel7_minimal]]], indirect=True)
+def test_v2v_ui_migration_plan_sorting(appliance, v2v_providers,
+        form_data_multiple_vm_obj_single_datastore, soft_assert):
+    infrastructure_mapping_collection = appliance.collections.v2v_mappings
+    migration_plan_collection = appliance.collections.v2v_plans
+
+    mapping = infrastructure_mapping_collection.create(
+        form_data_multiple_vm_obj_single_datastore[0])
+
+    plan1 = migration_plan_collection.create(
+        name="plan_{}".format(fauxfactory.gen_alphanumeric()), description="desc_{}"
+        .format(fauxfactory.gen_alphanumeric()), infra_map=mapping.name,
+        vm_list=[form_data_multiple_vm_obj_single_datastore[1][0]], start_migration=False)
+    plan2 = migration_plan_collection.create(
+        name="plan_{}".format(fauxfactory.gen_alphanumeric()), description="desc_{}"
+        .format(fauxfactory.gen_alphanumeric()), infra_map=mapping.name,
+        vm_list=[form_data_multiple_vm_obj_single_datastore[1][1]], start_migration=False)
+
+    view = navigate_to(migration_plan_collection, 'All')
+
+    plans_list_before_sort = view.migration_plans_not_started_list.read()
+    view.sort_direction.click()
+    plans_list_after_sort = view.migration_plans_not_started_list.read()
+    soft_assert(not (plans_list_before_sort == plans_list_after_sort))
+
+    view.migration_plans_not_started_list.schedule_migration(plan1.name, after_mins=5)
+    view.wait_displayed()
+    view.migration_plans_not_started_list.schedule_migration(plan2.name, after_mins=10)
+    view.wait_displayed()
+    view.sort_type_dropdown.item_select("Scheduled Time")
+    plans_list_before_sort = view.migration_plans_not_started_list.read()
+    view.sort_direction.click()
+    plans_list_after_sort = view.migration_plans_not_started_list.read()
+    soft_assert(not (plans_list_before_sort == plans_list_after_sort))
```

Comment:
Do you mean instead of `not` I should use `!=` ? If so, I really don't have an explanation on why. I just picked whatever came to my mind :) Do you think one is better than other?

Reply:
!= is better option in such case

Reply:
Can you help me understand the rationale. I did search a little bit on the internet to try to find what is recommended or preferred  and why. I did not find a good answer. Also, in our codebase, there are instances of `not` being used with `==` just like this, so not sure why they are allowed, but this one is not. Trying to expand my knowledge of the topic. Thanks.

Reply:
`!=`, `==` etc are comparator operators. Those are pointing to an equality test according to their `__eq__` or `__cmp__` methods.  This method return type is `bool`.

Now, In your our case, `A != B` will return us exact output what we expect. but we are doing   `A == B` it returning `bool` out and then with negation operator (`not`) we are trying to get what we expected.

```
In [1]: A = 1
In [2]: B = 2
In [3]: x = A == B
In [4]: x
Out[4]: False
In [5]: not(x)
Out[5]: True

In [6]: # If we have a facility and direct operator why not to use it?
In [7]: A!=B
Out[7]: True
```
As per my basic understanding
`!=` is `c_operator` and `not` is `py_operator` as `cpython` use `c` as base lang.

If you like to go in deep then let me know. I would like to point you `not` implementation from `CPython`.